pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -176,6 +176,9 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.CPUCFSQuotaPeriod == nil {
 		obj.CPUCFSQuotaPeriod = &metav1.Duration{Duration: 100 * time.Millisecond}
 	}
+	if obj.NodeStatusMaxImages == 0 {
```

Comment:
Could I have previously set the max images to 0 to make the node publish no images?  Is there a way we can preserve that and distinguish between 0 and not set?

Reply:
Nice catch for this corner case, moving this to a pointer can solve the problem:

1. If nil (not passed) a default value of 50 is used. - NodeStatusMaxImages:50,
2. If 0 is passed, none images comes from the slice. No .Node.Status.Image
3. If -1 is passed, containerImages list become unlimited.
4. If < -1 is passed, an invalid error is raised.